.github/workflows/test_generation.yaml

```diff
@@ -0,0 +1,77 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Test Library Generation
+# This workflow verifies the tooling that creates the library. It does this by
+# running the generator, and verifying that the tests pass on the
+# newly-generated library (not the library code currently committed)
+on:
+  pull_request:
+    paths:
+      - "scripts/**"
```

Comment:
just to capture the need here - the fallback workflow exists so that this check *name* can be made required, and satisfied by either the correct tests (for the paths in question) or the fallback test (for anything else) - Do I have that right?